src/ray/raylet/worker.h

```diff
@@ -74,33 +74,32 @@ class Worker {
   void WorkerLeaseGranted(const std::string &address, int port);
 
   // Setter, geter, and clear methods  for allocated_instances_.
-  void SetAllocatedInstances(TaskResourceInstances &allocated_instances) { 
+  void SetAllocatedInstances(std::shared_ptr<TaskResourceInstances> allocated_instances) { 
       allocated_instances_ = allocated_instances;
+  };                            
+
+  std::shared_ptr<TaskResourceInstances> GetAllocatedInstances() {return allocated_instances_; };    
+  
+  void ClearAllocatedInstances() { allocated_instances_ = nullptr; };
+
+  void SetLifetimeAllocatedInstances(std::shared_ptr<TaskResourceInstances> allocated_instances) { 
```

Comment:
```suggestion
  void SetLifetimeAllocatedInstances(std::shared_ptr<TaskResourceInstances> &allocated_instances) { 
```

This will avoid incrementing the shared pointer once, which can be expensive